Added hostapd and wpa_supplicant building

slyusar003 opened the pull request.

arnout committed:
tools: add hostapd.config file

This will be used for building hostapd.

For now, it is just the upstream defconfig with CONFIG_ACS enabled.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

arnout commented on tools/commands/build.py:
```diff
@@ -125,6 +128,33 @@ def make(self):
         subprocess.check_call(cmd, env=self.env)
 
+# custom builder class
+class cubuilder(builder):
```
This is really a hostap-specific builder, so call it `hostapbuilder` maybe.

arnout reviewed (changes requested):
One small change request.

When you resubmit, please do a rebase on master first.

Also, mention in the commit message what has changed w.r.t. the hostapd resp. wpa_supplicant defconfig. For hostapd, it is only CONFIG_ACS=y.

slyusar003 committed:
tools: Added hostapd and wpa_supplicant building

acbuilder changes directory. This happened to work before because
nothing else depends on the current directory. However, now the hostap
builder gets the config file from the current directory. Therefore,
change directory back to the original location at the end of
acbuilder.prepare().

hostapd.config and wpa_supplicant.config are equal to the upstream
defconfig but set CONFIG_ACS=y.

Signed-off-by: Ostap <slyusar.ostap@gmail.com>

arnout commented on tools/commands/build.py:
```diff
@@ -127,6 +132,33 @@ def make(self):
+    def __init__(self, name, modules_dir, install_dir, make_verbose=False):
+        self.make_verbose = make_verbose
+        super(hostapbuilder, self).__init__(name, modules_dir, modules_dir, install_dir)
+        self.config_src_fpath = os.getcwd() + "/config/" + name + ".config"
```
Don't use getcwd; use os.path.abspath:
```suggestion
        self.config_src_fpath = os.path.abspath(os.path.join("config", name + ".config"))
```

arnout commented on tools/commands/build.py:
```diff
@@ -127,6 +132,33 @@ def make(self):
+
+    def make(self):
+        bin_dir = "BINDIR=" + self.install_path + "/bin"
+        cmd = ["make", bin_dir, '-C', self.src_path, "install"]
```
I know they're the same, but use build_path here:
```suggestion
        cmd = ["make", bin_dir, '-C', self.build_path, "install"]
```

arnout commented on tools/commands/build.py:
```diff
@@ -137,28 +137,35 @@ class hostapbuilder(builder):
         subprocess.check_call(cmd, env=self.env)
 
+        # it requires to do a clean because the another builder instance with different config
+        # may use pre-built object files while building
```
Excellent explanation! Thanks!
